src/emmy/calculus/derivative.cljc

```diff
@@ -192,24 +229,66 @@
 ;; passed to the structure of functions, instead of separately for every entry
 ;; in the structure.
 ;;
+;; A dynamic variable controls whether or not this process uses forward-mode or
+;; reverse-mode AD.
+;;
 ;; TODO: I think this is going to cause problems for, say, a Structure of
 ;; PowerSeries, where there is actually a cheap `g/partial-derivative`
 ;; implementation for the components. I vote to back out this `::s/structure`
 ;; installation.
 
+(def ^:dynamic *mode* d/FORWARD-MODE)
+
 (doseq [t [::v/function ::s/structure]]
```

Comment:
The dynamic variable bothers me a tiny bit so I want to see if it's really necessary. I suspect it is, but here's one question. I can see having D-forward and D-reverse, and letting D be bound to one or the other by default. That would be enough, except we need to know which D partial differentiation implies, and we don't want the naming to mushroom into having two options for everything differentiation-like. But isn't there a clear choice when you want just one partial derivative out of many?